types/types.go

```diff
@@ -5,6 +5,7 @@ import (
 	"io"
 	"time"
 
+	"github.com/alexlarsson/tar-diff/pkg/tar-patch"
```

Comment:
`types.go` should be extremely cheap to use, without dragging in dependencies like compression implementations.

Reply:
The only real dependency on it is that we return a tar_patch.DataSource interface, which is a simple interface to look up stream data given a pathname. We can probably wrap this interface inside types.go to avoid the external dependency.